src/Service/Locker.php

```diff
@@ -2,69 +2,76 @@
 
 namespace App\Service;
 
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Connections\PrimaryReadReplicaConnection;
 use Doctrine\Persistence\ManagerRegistry;
 use App\Util\DoctrineTrait;
 
 class Locker
 {
     use DoctrineTrait;
 
-    private ManagerRegistry $doctrine;
-
-    public function __construct(ManagerRegistry $doctrine)
+    public function __construct(private ManagerRegistry $doctrine)
     {
-        $this->doctrine = $doctrine;
     }
 
-    public function lockPackageUpdate(int $packageId, int $timeout = 0)
+    public function lockPackageUpdate(int $packageId, int $timeout = 0): bool
     {
-        $this->getConn()->connect('master');
+        $this->connect();
 
         return (bool) $this->getConn()->fetchOne('SELECT GET_LOCK(:id, :timeout)', ['id' => 'package_update_'.$packageId, 'timeout' => $timeout]);
     }
 
-    public function unlockPackageUpdate(int $packageId)
+    public function unlockPackageUpdate(int $packageId): void
     {
-        $this->getConn()->connect('master');
+        $this->connect();
 
         $this->getConn()->fetchOne('SELECT RELEASE_LOCK(:id)', ['id' => 'package_update_'.$packageId]);
     }
 
-    public function lockSecurityAdvisory(string $source, int $timeout = 0)
+    public function lockSecurityAdvisory(string $source, int $timeout = 0): bool
     {
-        $this->getConn()->connect('master');
+        $this->connect();
 
         return (bool) $this->getConn()->fetchOne('SELECT GET_LOCK(:id, :timeout)', ['id' => 'security_advisory_'.$source, 'timeout' => $timeout]);
     }
 
-    public function unlockSecurityAdvisory(string $source)
+    public function unlockSecurityAdvisory(string $source): void
     {
-        $this->getConn()->connect('master');
+        $this->getConn()->connect();
 
         $this->getConn()->fetchOne('SELECT RELEASE_LOCK(:id)', ['id' => 'security_advisory_'.$source]);
     }
 
-    public function lockCommand(string $command, int $timeout = 0)
+    public function lockCommand(string $command, int $timeout = 0): bool
     {
-        $this->getConn()->connect('master');
+        $this->connect();
 
         return (bool) $this->getConn()->fetchOne(
             'SELECT GET_LOCK(:id, :timeout)',
             ['id' => $command, 'timeout' => $timeout]
         );
     }
 
-    public function unlockCommand(string $command)
+    public function unlockCommand(string $command): void
     {
-        $this->getConn()->connect('master');
+        $this->connect();
 
         $this->getConn()->fetchOne(
             'SELECT RELEASE_LOCK(:id)',
             ['id' => $command]
         );
     }
 
-    private function getConn()
+    private function connect(): void
+    {
+        $connection = $this->getConn();
+        if ($connection instanceof PrimaryReadReplicaConnection) {
+            $connection->connect('primary');
```

Comment:
this should use `$connection->ensureConnectedToPrimary()` instead of the legacy API of passing an argument to `connect`